### ISSUE TITLE
git clone error

### ISSUE THREAD

**Issue by DoveBoy** [git-for-windows/git#3377]
- [ ] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?

```
$ git --version --build-options

** git version 2.33.0.windows.2
cpu: x86_64
built from commit: 8735530946cced809cc6cc4c2ca3b078cdb3dfc8
sizeof-long: 4
sizeof-size_t: 8
shell-path: /bin/sh
feature: fsmonitor--daemon **
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

```
$ cmd.exe /c ver

** Microsoft Windows [Version 10.0.19043.1202] **
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

```
# One of the following:
> type "C:\Program Files\Git\etc\install-options.txt"
> type "C:\Program Files (x86)\Git\etc\install-options.txt"
> type "%USERPROFILE%\AppData\Local\Programs\Git\etc\install-options.txt"
$ cat /etc/install-options.txt

** use scoop install git command on my PC **
```

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

** insert your response here **

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

** windowsterminal-preview 1.10.1933.0 & windows-terminal 1.9.1942.0 **

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
** git clone https://gitee.com/babysor/Realtime-Voice-Clone-Chinese.git **
```
 - What did you expect to occur after running these commands?

** git clone the repository on my PC **

 - What actually happened instead?

** error! All repositories on gitee.com, both public and private, cannot be git cloned, the last version of git is fine **

![image](https://user-images.githubusercontent.com/19201447/130164476-b103fcdb-5afb-4e92-8e5b-a85517aa6719.png)

 - If the problem was occurring with a specific repository, can you provide the
   URL to that repository to help us with testing?

** https://gitee.com/babysor/Realtime-Voice-Clone-Chinese **

**Comment by cnmade** [comment]
This may because of the network timeout. try to clone with --depth=1 , it will clone for less data transfer. will have much more chance to success clone.

**Comment by DoveBoy** [comment]
> This may because of the network timeout. try to clone with --depth=1 , it will clone for less data transfer. will have much more chance to success clone.

But use `git pull` command also error

**Comment by dscho** [comment]
> the last version of git is fine

Which one is that? Could you pinpoint which of the [snapshots](https://wingit.blob.core.windows.net/files/index.html) is the first bad one?

**Comment by DoveBoy** [comment]
> > the last version of git is fine
> 
> Which one is that? Could you pinpoint which of the [snapshots](https://wingit.blob.core.windows.net/files/index.html) is the first bad one?

`Git-2.33.0-64-bit.exe` This vision is error git from `gitee.com`
`Git-2.32.0.2-64-bit.exe` This vision is normal

**Comment by DoveBoy** [comment]
@dscho 
[v2.32.0.windows.2](https://github.com/git-for-windows/git/releases/tag/v2.32.0.windows.2)This version is fine
![image](https://user-images.githubusercontent.com/19201447/130317835-4379c558-67b8-46ea-ab8e-dc4785accce6.png)

**Comment by dscho** [comment]
> 
> 
> @dscho
> [v2.32.0.windows.2](https://github.com/git-for-windows/git/releases/tag/v2.32.0.windows.2)This version is fine
> ![image](https://user-images.githubusercontent.com/19201447/130317835-4379c558-67b8-46ea-ab8e-dc4785accce6.png)

And the next one, built from https://github.com/git-for-windows/git/commit/37796bca76ef4180c39ee508ca3e42c0777ba444, is not? 
![image](https://user-images.githubusercontent.com/127790/130440792-1445c3f2-2782-4636-84d4-a72c6ed648a2.png)

Please clarify, your comment was not clear to me at all.

**Comment by DoveBoy** [comment]
@dscho This build from [3d45ac813c4adf97fe3733c1f763ab6617d5add5](https://github.com/git-for-windows/git/commit/3d45ac813c4adf97fe3733c1f763ab6617d5add5) working fine on Windows10 x64. But [fb59c1de9fd599a3eeb8459eb7cab731c6672ea3](https://github.com/git-for-windows/git/commit/fb59c1de9fd599a3eeb8459eb7cab731c6672ea3) not working fine, It can't to git clone `gitee.com`

**Comment by dscho** [comment]
> build from [3d45ac813c4adf97fe3733c1f763ab6617d5add5](https://github.com/git-for-windows/git/commit/3d45ac813c4adf97fe3733c1f763ab6617d5add5) working fine on Windows10 x64. But [fb59c1de9fd599a3eeb8459eb7cab731c6672ea3](https://github.com/git-for-windows/git/commit/fb59c1de9fd599a3eeb8459eb7cab731c6672ea3) not working fine

There are many snapshots in between. Which reflects many more commits that could be the culprit. If you expect me to work on this, so do I expect you to work on this, by pinpointing which of those snapshots is the first one that breaks for you.

**Comment by DoveBoy** [comment]
> There are many snapshots in between. Which reflects many more commits that could be the culprit. If you expect me to work on this, so do I expect you to work on this, by pinpointing which of those snapshots is the first one that breaks for you.

I use the `scoop install` [git](https://github.com/ScoopInstaller/Main/blob/master/bucket/git.json). This version not working fine.
![image](https://user-images.githubusercontent.com/19201447/130445271-b7d86fb6-6b15-4150-a11c-caa147abb1c4.png)

**Comment by dscho** [comment]
> I use the `scoop install` [git](https://github.com/ScoopInstaller/Main/blob/master/bucket/git.json).

That's not helpful. Please do download a portable Git somewhere in the middle of the snapshots you know to be good/broken, then test it. Then continue with the process until you find the first snapshot that broke it to you.

**Comment by DoveBoy** [comment]
@dscho After testing. Use `scoop install` [git](https://github.com/git-for-windows/git/releases/download/v2.33.0.windows.1/PortableGit-2.33.0-64-bit.7z.exe) not working fine. But use [snapshots](https://wingit.blob.core.windows.net/files/index.html) download [git](https://wingit.blob.core.windows.net/files/Git-2.33.0-64-bit.exe) working fine.
In fact, the two ways to download the git is not the same installation package, compared to the hash value is not the same. The problematic one should be the [PortableGit-2.33.0-64-bit.7z.exe](https://github.com/git-for-windows/git/releases/download/v2.33.0.windows.1/PortableGit-2.33.0-64-bit.7z.exe) installer.
![image](https://user-images.githubusercontent.com/19201447/130449149-9854f6ed-c064-43fb-84f3-a4724b0ab47a.png)

**Comment by dscho** [comment]
> In fact, the two ways to download the git is not the same installation package, compared to the hash value is not the same

That probably means that you're using different SSL backends because the portable Git defaults to OpenSSL but the installer asks you whether you want to use Secure Channel instead. But that information was not conveyed here, so I am forced to guess, which I really dislike.

**Comment by DoveBoy** [comment]
> portable Git defaults to OpenSSL

Previous version of [Portable Git](https://github.com/git-for-windows/git/releases/download/v2.32.0.windows.2/PortableGit-2.32.0.2-64-bit.7z.exe) working fine

**Comment by dscho** [comment]
> Previous version of [Portable Git](https://github.com/git-for-windows/git/releases/download/v2.32.0.windows.2/PortableGit-2.32.0.2-64-bit.7z.exe) working fine

I think I understand that you are unwilling to test the snapshots.

**Comment by DoveBoy** [comment]
> I think I understand that you are unwilling to test the snapshots.

Indeed 😅, because there are too many versions, need to download one by one to test, install and uninstall

**Comment by d1313113** [comment]
After my brew upgrade, I had the same problem.
Cannot clone, pull, push when using ssh in gitee.com, but http is fine.
I have tested that everything is ok when using 「 Sourcetree 」to operate via ssh.
I have tried to clone with --depth=1 or setting core.compression 0 and http.postBuffer, but not work.
```bash
$ git --version --build-options
git version 2.33.0
cpu: x86_64
no commit associated with this build
sizeof-long: 8
sizeof-size_t: 8
shell-path: /bin/sh
```
<img width="553" alt="Snipaste_2021-08-24_10-06-05" src="https://user-images.githubusercontent.com/35473748/130544294-d3fd5c35-bf64-4dc8-a22e-b0869db0f238.png">
<img width="506" alt="Snipaste_2021-08-24_10-03-40" src="https://user-images.githubusercontent.com/35473748/130544302-bf6802fa-ada4-427b-9a81-e93fc0e78ca0.png">

**Comment by hazendaz** [comment]
Having same issue with 2.33.0 and 2.33.0.2.  Dropping to 2.32.0.2 works.  Used the downloads in releases directly here so no snapshot versions.  Checked the settings based on ssh comment and that did not help (tried both using bundled and external).  Only issue I'm having is git clone fails.  For now, I'm just going to use 2.32.0.2 but wanted to comment that I'm facing same issue.

**Comment by DoveBoy** [comment]
> For now, I'm just going to use 2.32.0.2

Me too.I'm having is `git clone` and `git pull` fails

**Comment by dscho** [comment]
Without more detailed investigation by those affected (such as testing snapshot versions), it is highly unlikely that this issue can even start to get resolved.

Yes, it's a lot of work. Expecting _others_ to do that work _and_ then to identify the issue _and_ then to fix the bug _and_ then to release a fix _may_ turn out not to have the intended effect...

**Comment by hazendaz** [comment]
This is fixed in 2.33.1.
